add.go

```diff
@@ -655,3 +655,33 @@ func (b *Builder) userForCopy(mountPoint string, userspec string) (uint32, uint3
 	}
 	return owner.UID, owner.GID, nil
 }
+
+// EnsureContainerPathAs creates the specified directory owned by USER
+// with the file mode set to MODE.
+func (b *Builder) EnsureContainerPathAs(path, user string, mode *os.FileMode) error {
+	mountPoint, err := b.Mount(b.MountLabel)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err2 := b.Unmount(); err2 != nil {
+			logrus.Errorf("error unmounting container: %v", err2)
+		}
+	}()
+
+	uid, gid := uint32(0), uint32(0)
+	if user != "" {
+		if uidForCopy, gidForCopy, err := b.userForCopy(mountPoint, user); err == nil {
+			uid = uidForCopy
+			gid = gidForCopy
+		}
+	}
+
+	idPair := &idtools.IDPair{UID: int(uid), GID: int(gid)}
+	opts := copier.MkdirOptions{
+		ChmodNew: mode,
+		ChownNew: idPair,
```

Comment:
Pretty sure this needs to either supply a `UIDMap` and `GIDMap`, or handle ID mapping itself.

Reply:
I've taken it from the existing code (where I cannot spot any `UIDMap` and `GIDMap` conversion): https://github.com/containers/buildah/pull/3840/files#diff-f2e4566c6b7e38384283187aba6d7fd91e5ba8da2ffd0f849277bb76bff27fb3L1573

I've added it now

Reply:
It's entirely possible it wasn't ever actually creating the directory here.  If we don't have a test for that, we should add one.